crates/ruff_linter/src/rules/flake8_todos/rules/todos.rs

```diff
@@ -244,6 +248,7 @@ static ISSUE_LINK_TODO_LINE_REGEX_SET: LazyLock<RegexSet> = LazyLock::new(|| {
     RegexSet::new([
         r"\s*(http|https)://.*", // issue link
         r"\s*#\d+.*",            // issue code - like "#003"
+        r"\s*[A-Z]{2,}-\d+.*",   // Jira-style issue code - like "RFFU-6877"
```

Comment:
Looking through the ecosystem reports, I think we should make this more strict and has to be one of:

* Comes directly after todo (`TODO RFFU-6887` or `TODO: RFFU-6887)
* In parentheses `(RFFU-6877)`


@simonpercivall how do you format your Jira style issue codes that are easy to detect but don't lead to accidential false-negatives?

Reply:
@danparizher can you explain the your reasoning for why you only allow the codes at the end of a line, in parentheses, or after a colon but e.g. not when it's directly after a `TODO`?

Reply:
@MichaReiser The regex patterns are intentionally restrictive to prevent false positives. If we matched issue codes directly after "TODO" (like `# TODO PROJ-123`), we'd incorrectly flag legitimate descriptions that happen to mention issue numbers, such as `# TODO: Fix bug ABC-123` or `# TODO: Working on PROJ-123 and PROJ-456`.

By requiring issue codes the way they are currently implemented, we ensure they're deliberately placed as references rather than accidentally mentioned in descriptive text. This follows the principle that it's better to miss some edge cases than to generate incorrect warnings.